examples/libretro/tic80_libretro.c

```diff
@@ -0,0 +1,588 @@
+	state.keymap[tic_key_p] = RETROK_p;
+	state.keymap[tic_key_q] = RETROK_q;
+	state.keymap[tic_key_r] = RETROK_r;
+	state.keymap[tic_key_s] = RETROK_s;
+	state.keymap[tic_key_t] = RETROK_t;
+	state.keymap[tic_key_u] = RETROK_u;
+	state.keymap[tic_key_v] = RETROK_v;
+	state.keymap[tic_key_w] = RETROK_w;
+	state.keymap[tic_key_x] = RETROK_x;
+	state.keymap[tic_key_y] = RETROK_y;
+	state.keymap[tic_key_z] = RETROK_z;
+	state.keymap[tic_key_0] = RETROK_0;
+	state.keymap[tic_key_1] = RETROK_1;
+	state.keymap[tic_key_2] = RETROK_2;
+	state.keymap[tic_key_3] = RETROK_3;
+	state.keymap[tic_key_4] = RETROK_4;
+	state.keymap[tic_key_5] = RETROK_5;
+	state.keymap[tic_key_6] = RETROK_6;
+	state.keymap[tic_key_7] = RETROK_7;
+	state.keymap[tic_key_8] = RETROK_8;
+	state.keymap[tic_key_9] = RETROK_9;
+	state.keymap[tic_key_minus] = RETROK_MINUS;
+	state.keymap[tic_key_equals] = RETROK_EQUALS;
+	state.keymap[tic_key_leftbracket] = RETROK_LEFTBRACKET;
+	state.keymap[tic_key_rightbracket] = RETROK_RIGHTBRACKET;
+	state.keymap[tic_key_backslash] = RETROK_BACKSLASH;
+	state.keymap[tic_key_semicolon] = RETROK_SEMICOLON;
+	state.keymap[tic_key_apostrophe] = RETROK_QUOTE;
+	state.keymap[tic_key_grave] = RETROK_TILDE;
+	state.keymap[tic_key_comma] = RETROK_COMMA;
+	state.keymap[tic_key_period] = RETROK_PERIOD;
+	state.keymap[tic_key_slash] = RETROK_SLASH;
+	state.keymap[tic_key_space] = RETROK_SPACE;
+	state.keymap[tic_key_tab] = RETROK_TAB;
+	state.keymap[tic_key_return] = RETROK_RETURN;
+	state.keymap[tic_key_backspace] = RETROK_BACKSPACE;
+	state.keymap[tic_key_delete] = RETROK_DELETE;
+	state.keymap[tic_key_insert] = RETROK_INSERT;
+	state.keymap[tic_key_pageup] = RETROK_PAGEUP;
+	state.keymap[tic_key_pagedown] = RETROK_PAGEDOWN;
+	state.keymap[tic_key_home] = RETROK_HOME;
+	state.keymap[tic_key_end] = RETROK_END;
+	state.keymap[tic_key_up] = RETROK_UP;
+	state.keymap[tic_key_down] = RETROK_DOWN;
+	state.keymap[tic_key_left] = RETROK_LEFT;
+	state.keymap[tic_key_right] = RETROK_RIGHT;
+	state.keymap[tic_key_capslock] = RETROK_CAPSLOCK;
+	state.keymap[tic_key_ctrl] = RETROK_LCTRL;
+	state.keymap[tic_key_shift] = RETROK_LSHIFT;
+	state.keymap[tic_key_alt] = RETROK_LALT;
+	state.keymap[tic_key_escape] = RETROK_ESCAPE;
+	state.keymap[tic_key_f1] = RETROK_F1;
+	state.keymap[tic_key_f2] = RETROK_F2;
+	state.keymap[tic_key_f3] = RETROK_F3;
+	state.keymap[tic_key_f4] = RETROK_F4;
+	state.keymap[tic_key_f5] = RETROK_F5;
+	state.keymap[tic_key_f6] = RETROK_F6;
+	state.keymap[tic_key_f7] = RETROK_F7;
+	state.keymap[tic_key_f8] = RETROK_F8;
+	state.keymap[tic_key_f9] = RETROK_F9;
+	state.keymap[tic_key_f10] = RETROK_F10;
+	state.keymap[tic_key_f11] = RETROK_F11;
+	state.keymap[tic_key_f12] = RETROK_F12;
+	state.keymap[tic_key_f12] = RETROK_F12;
+	state.keymap[tic_keys_count] = RETROK_LAST;
+}
+
+/**
+ * libretro callback; Global deinitialization.
+ */
+void retro_deinit(void)
+{
+	// Try to force another unload of the game.
+	retro_unload_game();
```

Comment:
why though? this is guaranteed to be called before deinit

Reply:
Sweet! Thanks.